### PR TITLE
test: Terraform local start-api integration tests base

### PR DESCRIPTION
**Pull request by mildaniel**
#### Which issue(s) does this change fix?

N/A

#### Why is this change necessary?
Add integration testing logic for new Terraform local start-api feature.

#### What side effects does this change have?
N/A

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/integration/local/start_api/test_start_api_with_terraform_application.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        except (FileNotFoundError, PermissionError):
+            pass
+
+
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+@pytest.mark.flaky(reruns=3)
+class TestStartApiTerraformApplication(TerraformStartApiIntegrationBase):
```
Comment (lucashuy) [inline tests/integration/local/start_api/test_start_api_with_terraform_application.py:46]: Did we usually set a pytest timeout for these tests? Or did we usually rely on the `start-api` command to time us out?

**Review by lucashuy — COMMENTED** [review]
LGTM in general, just left a question

File: tests/integration/local/start_api/test_start_api_with_terraform_application.py (hunk 1 above)
Comment (mildaniel) [reply tests/integration/local/start_api/test_start_api_with_terraform_application.py:46]: We have a timeout configured in the `wait_for_local_process` function that starts up the process in the tests. Is that what you're referring to?

https://github.com/aws/aws-sam-cli/blob/develop/tests/integration/local/common_utils.py#L8

Comment (lucashuy) [reply tests/integration/local/start_api/test_start_api_with_terraform_application.py:46]: More so with after the API loads and we make a request + wait for response

Comment (lucashuy) [reply tests/integration/local/start_api/test_start_api_with_terraform_application.py:46]: Oh scratch what I said, the request has it's own timeout defined 😅

**Review by moelasmar — APPROVED** [review]

**Pull request merged**